### PR TITLE
changed cryptos-rankings table

### PR DESCRIPTION
**Pull request by lawrenceslng**
updated cryptos-rankings to show deals and transaction instead of venues and transactions

### CODE REVIEW

File: routes/cryptos_ranking.js [hunk 1]
```diff
@@ -20,10 +21,23 @@ router.get('/api/cryptosranking_venues', function(req, res) {
...
+    }
+  );
+});
+
 router.get('/api/cryptosranking_transactions', function(req, res) {
+  //this is taking paypal transactions into account, making query return a null row
+  //added condition WHERE crypto_id is not null
   connection.query(
-    'SELECT users_purchases.crypto_id, crypto_metadata.crypto_name, crypto_metadata.crypto_symbol, crypto_info.crypto_logo, crypto_metadata.crypto_price, count(users_purchases.crypto_id) as total_transactions FROM users_purchases LEFT JOIN crypto_metadata ON users_purchases.crypto_id = crypto_metadata.id LEFT JOIN crypto_info ON crypto_metadata.crypto_name = crypto_info.crypto_metadata_name GROUP BY users_purchases.crypto_id ORDER BY total_transactions DESC',
+    'SELECT up.crypto_id, cm.crypto_name, cm.crypto_symbol, ci.crypto_logo, cm.crypto_price, count(up.crypto_id) as total_transactions FROM users_purchases up LEFT JOIN crypto_metadata cm ON up.crypto_id = cm.id LEFT JOIN crypto_info ci ON cm.crypto_name = ci.crypto_metadata_name WHERE up.crypto_id IS NOT NULL GROUP BY up.crypto_id ORDER BY total_transactions DESC',
```
Comment (simonnguyen3054) [inline routes/cryptos_ranking.js]: @lawrenceslng this is interesting. I didn't know that you can abbreviate the table names.

Comment (avakrishn) [reply routes/cryptos_ranking.js]: @simonnguyen3054 , @lawrenceslng is using aliases which he has made as abbreviation of the table names similar to what we have done with other routes

File: routes/cryptos_ranking.js [hunk 2]
```diff
@@ -20,10 +21,23 @@ router.get('/api/cryptosranking_venues', function(req, res) {
...
+    function(err, deals_count, fields) {
+      if(err) console.log(err);
+      res.json(deals_count);
+    }
+  );
+});
+
 router.get('/api/cryptosranking_transactions', function(req, res) {
+  //this is taking paypal transactions into account, making query return a null row
+  //added condition WHERE crypto_id is not null
```
Comment (avakrishn) [inline routes/cryptos_ranking.js]: @lawrenceslng So if the crypto_id is null is that when there are no transactions for the crypto and thus will not be shown in the crypto ranking table?

Comment (lawrenceslng) [reply routes/cryptos_ranking.js]: the problem at first was that the original route took into account all transactions including paypal, and when it returns the total transactions, paypal shows up as a "null" crypto and thus it's displayed in the cryptorankings table even tho it shouldn't, so I merely added a condition where it checks each transaction record to see if it was completed with crypto or paypal; if crypto then it's all good, if it's paypal then we do not count that transaction in the cryptorankings by transaction table

Comment (avakrishn) [reply routes/cryptos_ranking.js]: Thanks @lawrenceslng

**Pull request merged**